app/controllers/solidus_stripe/intents_controller.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'stripe'
+
+class SolidusStripe::IntentsController < Spree::BaseController
+  include Spree::Core::ControllerHelpers::Order
+
+  before_action :load_payment_method
+
+  def payment_confirmation
+    payment = @payment_method.find_in_progress_payment_for(current_order)
+    intent = @payment_method.find_intent_for(payment)
+
+    if params[:payment_intent] != intent.id
+      raise "The payment intent id doesn't match"
+    end
+
+    unless %w[confirm payment].include?(current_order.state.to_s)
+      redirect_to main_app.checkout_state_path(current_order.state)
+      return
+    end
+
+    current_order.state = :payment
+    payment.log_entries.create!(details: { payment_intent: intent.to_json })
+
+    # https://stripe.com/docs/payments/intents?intent=payment
+    case intent.status
```

Comment:
As we agreed on handling everything with the webhooks, what about making this PR more straightforward by removing all the logic from here and simply displaying a message? That would also play better with the current test coverage.

Reply:
I'd like to keep the complete checkout flow to be working with proper payment status updates, and until we have webhooks this has served this function well. It's surely intended as nothing more than a draft of what the webhook response will be.

Thoughts on leaving it as it is or at least keeping what's required for the checkout spec to keep functioning?

Reply:
👍 Created #190